### PR TITLE
Removed 'data-newsletter' attribute from template because it's not doing anything

### PR DESCRIPTION
**Pull request by mmmavis**
TL;DR 
Removed `data-newsletter` attribute from template code because it's not doing anything. Please see below for detailed explanation.

Related PRs/issues: #11712

---

# Back end

When handling newsletter signup info, [backend](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/main/network-api/networkapi/campaign/views.py#L78) gets the `newsletters` value from the Signup object directly and not through the payload it receives. This means payload does not need to include such value.

# Front end

In the codebase, there are two ways to render newsletter signup modules:

### 1. Through `join-us.js` (code is being refactored and going away soon)

[`join-us.js`](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/main/source/js/common/inject-react/join-us.js#L24) renders the [newsletter module (join.jsx)](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/main/source/js/components/join/join.jsx) at where the `.join-us` DOM nodes are.

`data-newsletter` is passed as `props` into the React components `<JoinUs>`. However, in `join.jsx` implementation, looking for keywords like `newsletter` and `props`, I don't see the value of `newsletter` being used or manipulated anywhere. The payload in [submitDataToApi method](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/main/source/js/components/join/join.jsx#L133-L180) does not include `newsletter` info either. Hence, I've removed the `data-newsletter` attribute from `.join-us` DOM nodes.

**Link to sample test page**
- Create a BannerPage (BanneredCampaignPage) locally and link it with a Signup snippet.
- template code: https://github.com/MozillaFoundation/foundation.mozilla.org/pull/11724/files#diff-45e83b057173ef31d57fa50a5144e65fe24cb5ec6c3309d603057610aad21599
- when testing, please use `mozilla-foundation`, `mozilla-festival`, or `mozilla-foundation,mozilla-festival` as the "newsletter" value because Basket only accepts these.

### 2. Through `newsletter-signup-module.js`

[newsletter-signup-module.js](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/main/source/js/common/inject-react/newsletter-signup-module.js#L31) renders the [newsletter modules (default-layout-signup.jsx)](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/main/source/js/components/newsletter-signup/organisms/default-layout-signup.jsx) at where the `.newsletter-signup-module` DOM nodes are.

`data-newsletter` is passed as `props` into the React components `<DefaultLayoutSignup>`. However, in `default-layout-signup.jsx` and [its HOC component](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/main/source/js/components/newsletter-signup/organisms/with-submission-logic.jsx)'s implementation, looking for keywords like `newsletter` and `props`, I don't see the value of `newsletter` being used or manipulated anywhere. The payload in [submitDataToApi method](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/main/source/js/components/newsletter-signup/organisms/with-submission-logic.jsx#L169-L176) does not include `newsletter` info either. Hence, I've removed the `data-newsletter` attribute from `.newsletter-signup-module` DOM nodes.

**Link to sample test page**
- http://mozfest.localhost:8000/ (signup form on footer)
- template code: https://github.com/MozillaFoundation/foundation.mozilla.org/pull/11724/files#diff-ae5210abd937cbe7ed10c81b534e646ae8a820a29bc4ebff362a50352e8aa76e
- when testing, please use `mozilla-foundation`, `mozilla-festival`, or `mozilla-foundation,mozilla-festival` as the "newsletter" value because Basket only accepts these.

### CODE REVIEW

**Review by jhonatan-lopes — APPROVED** [review]
LGTM :+1:
